lib/pbench/test/unit/server/test_cache_manager.py

```diff
@@ -407,12 +417,23 @@ def generate_test_result_tree(tmp_path: Path, dir_name: str) -> Path:
                 parents=True, exist_ok=True
             )
             (sub_dir / "subdir1" / "subdir14" / "subdir141" / "f1411.txt").touch()
-            sym_file = sub_dir / "subdir1" / "subdir14" / "subdir141" / "f1412_sym"
-            os.symlink((sub_dir / "subdir1" / "f11.txt"), sym_file)
+            sym_file = sub_dir / "subdir1" / "f12_sym"
+            os.symlink(Path("..") / ".." / dir_name, sym_file)
             sym_file = sub_dir / "subdir1" / "subdir12" / "f121_sym"
             os.symlink((sub_dir / "subdir1" / "subdir15"), sym_file)
+            sym_file = sub_dir / "subdir1" / "subdir12" / "f122_sym"
+            os.symlink((Path(".") / "bad_subdir" / "nonexistent_file.txt"), sym_file)
             sym_file = sub_dir / "subdir1" / "subdir13" / "f131_sym"
             os.symlink((Path("/etc") / "passwd"), sym_file)
+            sym_file = sub_dir / "subdir1" / "subdir14" / "subdir141" / "f1412_sym"
+            os.symlink((sub_dir / "subdir1" / "f11.txt"), sym_file)
```

Comment:
FWIW, you don't need the outermost parentheses around the first argument in these calls to `os.symlink()`.